server/routes/updateGoal/updateGoalFormValidator.ts

```diff
@@ -0,0 +1,20 @@
+import type { UpdateGoalForm } from 'forms'
+import formatErrors from '../errorFormatter'
+import validateGoalTitle from '../../validators/goalTitleValidator'
+import validateGoalStatus from '../../validators/goalStatusValidator'
+import validateStepTitle from '../../validators/stepTitleValidator'
+import validateStepTargetDateRange from '../../validators/stepTargetDateRangeValidator'
+import validateStepStatus from '../../validators/stepStatusValidator'
+
+export default function validateUpdateGoalForm(updateGoalForm: UpdateGoalForm): Array<Record<string, string>> {
+  const errors: Array<Record<string, string>> = []
+
+  errors.push(...formatErrors('title', validateGoalTitle(updateGoalForm.title)))
+  errors.push(...formatErrors('status', validateGoalStatus(updateGoalForm.status)))
+  updateGoalForm.steps.forEach((step, idx) => {
+    errors.push(...formatErrors(`steps[${idx}].title`, validateStepTitle(step.title)))
```

Comment:
This is slightly different to how I've done ids/names on the form fields on my PR - may need to converge!